src/movegen/rooks.h

```diff
@@ -80,6 +86,7 @@ constexpr uint64_t rookAttacksWithBlock(int square, uint64_t block)
     return attacks;
 }
 
+#ifdef USE_BMI2
```

Comment:
I think that it would be cleaner if we could use as few of these checks in this file and move as much of the specifics into header files that we then include.

So eg:

```cpp
#ifdef PEXT_AVAILABLE
  #include "pext_magics.h" // or what name you prefer
#else
  #include "hashing_magics.h" // same here
#endif 
```